appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/YaFormEditor.java

```diff
@@ -1044,6 +1044,8 @@ function copy(e) {
     $wnd.addEventListener('keydown', function(e) {
       if (e.keyCode === 16) {
         editor.shiftDown = true;
+      } else if (e.keyCode == 46) {
+        editor.@com.google.appinventor.client.editor.youngandroid.YaFormEditor::deleteSelectedComponent()();
```

Comment:
This check is much too generous. For example, if I am focused in a property editor and press the delete key to delete a character it will prompt to ask whether I want to delete the component. Likewise, if I switch to another view like the blocks view or go back to the project view. You will need to have a function that checks whether the form's editor is currently active and that the focus is on a component and not an editor.